### ISSUE TITLE
[New build - EXPERIMENTAL] 22w38c: Children Entities and Inventory, Hotfixes

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#4932]
Mechanism are now able to connect to pipe, allowing inventories to be shared with children entities.

Bugfixes:
#4905 #4889 Some bricks do not allow pipe overlap.
#4901 Save corruption
#4930 Silver ingot labelling
#4920 Keybind quantity multiplier not working with Codex
#4430 Brick description cleanup.
- Quit to title and reload is broken
#4925 Tank size on old builds have shrunk.
#4926 Crashes during heavy mining.

**Comment by tsunamayo** [comment]
Build is up. As expected quite a rocky release, I will sort out everything!

**Comment by PopkaAkula** [comment]
Will we ever see the return of hoses as an alternative method to do cross entity inventory?

**Comment by Womble-TC** [comment]
I agree, hoses would be a really good solution for having cross entity fuel connection for small mechanisms like robot arms and such (if possible)

**Comment by CaptainKronos** [comment]
And hoses would just look really cool. Though, I would not be surprised if they are a bit physics intensive due to their nature as a hose.

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo I would love for mechanisms to be able to transport fuel as well!

**Comment by tsunamayo** [comment]
@Kaiser-Indrasil Yep this is planned, btw I will ditch the current fuel implementation and use the item logic to transport fuel. It should not impact your build.
@Womble-TC @PopkaAkula @CaptainKronos Yes I will try to work on that at one point! Cheers

**Comment by ProPeach** [comment]
Thanks for allowing overlap on some additional Construction type bricks as well as #4905! That was a very nice stealth QoL feature heh

**Comment by tsunamayo** [comment]
![image](https://user-images.githubusercontent.com/6604318/192416906-8a1fb531-a269-4617-817b-ee912d37b457.png)

@ProPeach @Womble-TC @PopkaAkula @Kaiser-Indrasil

**Comment by ZachZent** [comment]
hos

**Comment by Womble-TC** [comment]
These look perfect! Would they be for fuel or items, or both?

**Comment by tsunamayo** [comment]
@Womble-TC I will try to make a squarish one for item, lets see how it will look (I fear it might distord).

**Comment by Servomotor** [comment]
WHOA....bendy hoses?!?!!?!?!!?!

BENDY HOSES?!?!?!!?!?

**BENDY HOSES?!?!?!!?!?**

**Comment by tsunamayo** [comment]
@Servomotor time to flex those builds ^^

**Comment by Servomotor** [comment]
> @Servomotor time to flex those builds ^^

LOL...you are a goon. LOL

I am working on it :)

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo 
YOOOOOOO
HOOOOOOSE
Can't wait to use them as decorative elements!
> I will try to make a squarish one for item, lets see how it will look (I fear it might distord).

I would LOVE if you made a hose without ribs with a smooth texture instead. Maybe that could be a second model for inventory systems?

**Comment by Kaiser-Indrasil** [comment]
I have a few QA questions about the hoses:
- can they make an S shape? And how do they look if so?
- Can we daisy-chain the hoses together?

**Comment by ProPeach** [comment]
Aaaah what a tease! :D

**Comment by TIKIRobo** [comment]
Will we be able to edit some of the properties of the hos? Stuff like extra length, rigitity, etc

**Comment by pinesh** [comment]
Wooo! Hose's are making a comeback! glad to see it!

**Comment by tsunamayo** [comment]
@Kaiser-Indrasil 
Hoh S shape are looking _fantastique_
![image](https://user-images.githubusercontent.com/6604318/192656663-90b444b0-a32d-4f0e-b99d-72f361480466.png)

No you cant daisy-chain if you want to hose to start from the same side, but you can place two back-to-back.

I will make a smooth one, a glow one, and maybe a transparent one for fuel, but this will have to wait for the fuel refactor.
Cheers

**Comment by Servomotor** [comment]
Will the hoses need to be placed to a functional block, or can they be just for decoration?

**Comment by ProPeach** [comment]
I suppose you could say there are some _kinks_ to iron out for the S bend ; )

**Comment by ZachZent** [comment]
This is such a minor suggestion that it doesn't really matter, but when connecting the hoses, could the animation snake out from one or both sides instead of just appearing?

~~totally not so I can add a slurping sound~~

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo Fantastique news!
I am genuinely hyped for them!

**Comment by tsunamayo** [comment]
@Servomotor So no you cannot color the bendy part separately, actually you cant interact with it (there is no physics).
Then yes you can use for decoration like the other pipe/conveyor tubes.
Cheers all

**Comment by Uncle-Ulty** [comment]
You know that we gonna use hoses as greeble, don't you? 

besides that, I loved this new feature. But they gonna look super ugly if someone connects it to a piston or slider CE... the hose gonna extend for lots and lots of meters...

**Comment by tsunamayo** [comment]
@Uncle-Ulty yep we will see, at one point the builder has the responsibility to make it look good (by not using it if it looks bad). Also this introduces some game balance issue as these could be used inside systems, require less crafting materials ect.

In other news talking about umbilical's stuff I just got a second baby yesterday, cheers guys!

**Comment by pinesh** [comment]
Congratulations!

**Comment by ProPeach** [comment]
Oh wow, congratulations! I hope your wife is doing well too!

**Comment by ZachZent** [comment]
Congratulations! Another dev to work on the game!

**Comment by cyberjasse** [comment]
Congratulations Tsuna !

**Comment by Kaiser-Indrasil** [comment]
Congrats to you and the missus!

I hope her and the baby are well and healthy <3
